lstchain/tools/lstchain_create_dl2_file.py

```diff
@@ -0,0 +1,188 @@
+        ("o", "output"): "ReconstructionHDF5Writer.output_dir",
+        "energy-model": "ReconstructionHDF5Writer.path_energy_model",
+        "disp-model": "ReconstructionHDF5Writer.path_disp_model",
+        "gh-model": "ReconstructionHDF5Writer.path_gh_model",
+    }
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        """
+        Tool that generates a HDF5 file with reconstructed energy, disp and gammaness of events.
+
+        For getting help run:
+        lstchain_create_dl2_file --help
+        """
+        self.data_ind = None
+        self.data_src_dep = None
+        self.dl2_ind = None
+        self.dl2_src_dep_level = None
+        self.dl2_src_dep_dict = {}
+        self.reg_energy = None
+        self.reg_disp_vector = None
+        self.cls_gh = None
+
+        self._dict_conf = replace_config(standard_config, self.get_current_config())
+
+    def setup(self):
+
+        if self.config_file is None:
+            raise ToolConfigurationError("Please provide configuration file with --config parameter")
+
+        self.log.info("Reading DL1 file")
+        self.data_ind = pd.read_hdf(self.input, key=dl1_params_lstcam_key)
+        if self._dict_conf["source_dependent"]:
+            self.data_src_dep = pd.read_hdf(self.input, key=dl1_params_src_dep_lstcam_key)
+        self.log.info("Reading RF models")
+        self.reg_energy = joblib.load(self.path_energy_model)
+        self.reg_disp_vector = joblib.load(self.path_disp_model)
+        self.cls_gh = joblib.load(self.path_gh_model)
+
+    def start(self):
+
+        self.data_ind = add_delta_t_key(self.data_ind)
+        # dealing with pointing missing values
+        # this happened when `ucts_time` was invalid
+        if (
+            "alt_tel" in self.data_ind.columns
+            and "az_tel" in self.data_ind.columns
+            and (np.isnan(self.data_ind.alt_tel).any() or np.isnan(self.data_ind.az_tel).any())
+        ):
+            # make sure there is a least one good pointing value to interpolate from
+            if np.isfinite(self.data_ind.alt_tel).any() and np.isfinite(self.data_ind.az_tel).any():
+                self.data_ind = impute_pointing(self.data_ind)
+            else:
+                self.data_ind.alt_tel = -np.pi / 2.0
+                self.data_ind.az_tel = -np.pi / 2.0
+
+        # source-independent analysis
+        self.log.info("Applying models")
+        if not self._dict_conf["source_dependent"]:
+            self.data_ind = filter_events(
+                self.data_ind,
+                filters=self._dict_conf["events_filters"],
+                finite_params=self._dict_conf["regression_features"] + self._dict_conf["classification_features"],
+            )
+            self.dl2_ind = dl1_to_dl2.apply_models(
+                self.data_ind,
+                self.cls_gh,
+                self.reg_energy,
+                self.reg_disp_vector,
+                custom_config=self._dict_conf,
```

Comment:
If you want to be minimally invasive, construct `custom_config` from the traitlets here, but the better approach would be to implement a `Component` for `apply_models`.